x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleType.java

```diff
@@ -159,6 +159,11 @@ public static boolean shouldInjectMigrateStepForPhase(Phase phase) {
             return false;
         }
 
+        // do not inject if MigrateAction is not supported for this phase (such as hot, frozen, delete phase)
+        if (ALLOWED_ACTIONS.get(phase.getName()).contains(MigrateAction.NAME) == false) {
+            return false;
+        }
+
```

Comment:
Good point, I've updated the code in this way.

Reply:
What do you think about restructuring the conditionals like this:
- We start the method with this:
```
if (ALLOWED_ACTIONS.containsKey(phase.getName()) == false) {
    throw new IllegalArgumentException("Timeseries lifecycle does not support phase [" + phase.getName() + "]");
}
```
This way we make sure that we have consistent behaviour every time there is an invalid phase. The way the code is now, if there are searchable snapshots in an invalid phase it will return `false` instead of `IllegalArgumentException`.

- Then we could continue listing the phase independent cases, and finally just return if it's allowed in this phase or not:
```
// if the user configured the {@link MigrateAction} already we won't automatically configure it
if (phase.getActions().get(MigrateAction.NAME) != null) {
    return false;
}
        
// only inject the MigrateAction id it's supported for this phase (for example warm, cold).
return ALLOWED_ACTIONS.get(phase.getName()).contains(MigrateAction.NAME);
```

What do you think?

Reply:
Hm, I see, you are right we are changing the behaviour but I would expect if an invalid phase is provided that it would blow up anyway in a similar way since other methods do that too. I would like to ask another opinion, @dakrone do you see any red flags for throwing an `IllegalArgumentException` if we get an invalid phase?

Reply:
But in that case we would be throwing an `NPE`, right?

Reply:
@gmarouli that's correct. I missed we're actually adding the check that uses `ALLOWED_ACTIONS`. 

I think it makes sense to have a check to avoid NPE. IMO returning `false` for an invalid phase would be preferable - ie. migrate should NOT be injected (as we're not really validating the allowed phases here), but an exception could work too.

Reply:
Great, I am okay with returning `false` consistently when we receive a phase we do not recognize. @mushao999 if you agree too with this approach, let's replace the exception with returning false and let's ship it! 🚀

Reply:
Thanks! I will merge it with master asap.